src/commands.c

```diff
@@ -2309,7 +2310,7 @@ static RESPONSECODE CmdXfrBlockTPDU_T1(unsigned int reader_index,
 RESPONSECODE SetParameters(unsigned int reader_index, char protocol,
 	unsigned int length, unsigned char buffer[])
 {
-	unsigned char cmd[10+length+2];	/* CCID + Protocol Data Structure */
+	unsigned char cmd[ROUND_UP_BUF(10+length+2)];	/* CCID + Protocol Data Structure */
```

Comment:
I am wary of using VLAs (see e.g. [here](https://lwn.net/Articles/749064/)). It might be better to use malloc. Though I checked and in this case all actual `length` arguments are small constant size, so it's not exploitable or such.